launcher/internal/oci/cosign/signature.go

```diff
@@ -0,0 +1,64 @@
+// Package cosign contains functionalities to interact with signatures generated by cosign.
+// https://github.com/sigstore/cosign/blob/main/specs/SIGNATURE_SPEC.md.
+package cosign
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/google/go-tpm-tools/launcher/internal/oci"
+	"github.com/opencontainers/go-digest"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// Sig implements oci.Signature interface for cosign-gernated signatures.
+type Sig struct {
+	// Layer represents a layer descriptor for OCI image manifest.
+	Layer v1.Descriptor
+	// Blob represents the opaque data uploaded to OCI registory associated with the layer.
+	Blob []byte
+}
+
+// CosignSigKey is the key of the cosign-generated signature embedded in OCI image manifest.
+const CosignSigKey = "dev.cosignproject.cosign/signature"
+
+var (
+	// Verify that our Sig struct implements the expected public interface.
+	_        oci.Signature = Sig{}
+	encoding               = base64.StdEncoding
+)
+
+// Payload implements oci.Signature interface.
+func (s Sig) Payload() ([]byte, error) {
+	if digest.FromBytes(s.Blob) != s.Layer.Digest {
```

Comment:
Comparing digests, probably a good idea to use:  https://pkg.go.dev/crypto/subtle#ConstantTimeCompare

Reply:
Thanks for pointing me to the materials related to timing attack. I agree with your point because the layer digest is public visible from OCI image manifest so directly comparing these two values should be ok. Added more comments here.